xbmc/utils/StringUtils.cpp

```diff
@@ -1284,3 +1284,29 @@ void StringUtils::Tokenize(const std::string& input, std::vector<std::string>& t
     dataPos = input.find_first_not_of(delimiter, nextDelimPos);
   }
 }
+
+uint64_t StringUtils::ToUint64(std::string str, uint64_t fallback) noexcept
+{
+  std::istringstream iss(str);
+  uint64_t result(fallback);
+  iss >> result;
+  return result;
+}
+
+std::string StringUtils::FormatFileSize(uint64_t bytes)
+{
+  const std::array<std::string, 5> units{"B", "kB", "MB", "GB", "TB"};
+  if (bytes < 1000)
+    return Format("%luB", bytes);
+
+  int i = 0;
+  double value(bytes);
+  while (i < units.size() && value >= 999.5)
+  {
+    ++i;
+    value = bytes / std::pow(1024.0, i);
```

Comment:
@stefansaraev yes, 999.5. this is a smarter rounding that scales decimals with significance. See tests.